pkg/reconciler/managed/reconciler.go

```diff
@@ -510,6 +511,19 @@ func NewReconciler(m manager.Manager, of resource.ManagedKind, o ...ReconcilerOp
 	return r
 }
 
+// maybeBackoff updates the managed resource status and returns an error in the
+// case that backoff is enabled and the passed error is not nil. Otherwise, the
+// request is requeued with specified wait time, and the error result of the
+// update operation is returned. If the update operation error result is not
+// nil, backoff will also be triggered implicitly.
+func (r *Reconciler) maybeBackoff(ctx context.Context, m resource.Managed, wait time.Duration, err error) (reconcile.Result, error) {
+	updateErr := errors.Wrap(r.client.Status().Update(ctx, m), errUpdateManagedStatus)
+	if r.useBackoff && err != nil {
+		return reconcile.Result{}, err
```

Comment:
We could concatenate the update error in the case where we both pass an error, backoff is enabled, and the status update fails.